app/controllers/groups_controller.rb

```diff
@@ -136,9 +136,14 @@ def landing
 
   # DELETE /groups/:gid
   def destroy
-    @group.destroy
-    flash[:success] = "Group has been destroyed successfully."
-    head :no_content
+    if @group.notebooks.count == 0
+      message = "Group #{@group.name} has been deleted successfully."
+      @group.destroy
+      flash[:success] = message
```

Comment:
I feel like we should add a JSON response also to this on success that uses the message but I know this will be covered eventually by the one ticket to refactor all of that and ensure our system is returning the messages both in the UI and outside of it.

Reply:
No point in doing a JSON return on success because you have to navigate away from the group's page since it doesn't exist anymore, so the flash message suffices as it navigates to the group listing again.